File: apps/web/lib/types/inventory.ts
// ── Inventory view ─────────────────────────────────────────────────────────────

export interface InventoryEmployeeSummary {
  id: string
  first_name: string
  last_name: string
  role: string
}

export interface InventoryAssetItem {
  id: string
  asset_type: string
  name: string
  quantity: number
  unit: string | null
  serial_number: string | null
  notes: string | null
  assigned_at: string
}

export interface InventoryMaterialItem {
  project_material_id: string
  material_name: string
  material_code: string | null
  project_id: string
  project_name: string
  total_quantity: number
  unit: string
}

export interface InventoryResponse {
  employee: InventoryEmployeeSummary
  assets: InventoryAssetItem[]
  materials: InventoryMaterialItem[]
}

// ── Transfer targets ───────────────────────────────────────────────────────────

export interface TransferTarget {
  id: string
  first_name: string
  last_name: string
  role: string
}

// ── Employee active projects ───────────────────────────────────────────────────

export interface InventoryProject {
  id: string
  name: string
}

// ── Create transfer ────────────────────────────────────────────────────────────

export interface CreateTransferRequest {
  to_employee_id: string
  type: 'asset' | 'material'
  employee_asset_id?: string
  project_material_id?: string
  destination_project_id?: string
  quantity?: number
  notes: string
}

// ── Transfer list ──────────────────────────────────────────────────────────────

export interface TransferListItem {
  id: string
  from_employee_id: string
  from_employee_name: string
  to_employee_id: string
  to_employee_name: string
  asset_type: string
  item_name: string
  quantity: number
  transferred_at: string
  notes: string | null
  transferred_by_name: string
}

export interface TransferPagination {
  page: number
  per_page: number
  total: number
  total_pages: number
}

export interface TransferListResponse {
  data: TransferListItem[]
  pagination: TransferPagination
}

// ── UI helpers ─────────────────────────────────────────────────────────────────

export const ASSET_TYPE_LABELS: Record<string, string> = {
  car: 'Vozilo',
  tool: 'Alat',
  equipment: 'Oprema',
  material: 'Materijal',
  other: 'Ostalo',
}

export const ROLE_LABELS: Record<string, string> = {
  direktor: 'Direktor',
  inzenjer: 'Inženjer',
  administracija: 'Administracija',
  poslovoda: 'Poslovoda',
  radnik: 'Radnik',
}

export function canViewOtherInventory(role: string): boolean {
  return ['direktor', 'inzenjer', 'administracija'].includes(role)
}

export function canTransfer(role: string): boolean {
  return ['direktor', 'inzenjer', 'administracija', 'poslovoda'].includes(role)
}

export function employeeFullName(emp: InventoryEmployeeSummary): string {
  return `${emp.first_name} ${emp.last_name}`
}

File: services/api/dto/inventory.go
package dto

import "time"

// ── Inventory view ─────────────────────────────────────────────────────────────

type InventoryEmployeeSummary struct {
	ID        string `json:"id"`
	FirstName string `json:"first_name"`
	LastName  string `json:"last_name"`
	Role      string `json:"role"`
}

type InventoryAssetItem struct {
	ID           string    `json:"id"`
	AssetType    string    `json:"asset_type"`
	Name         string    `json:"name"`
	Quantity     float64   `json:"quantity"`
	Unit         *string   `json:"unit"`
	SerialNumber *string   `json:"serial_number"`
	Notes        *string   `json:"notes"`
	AssignedAt   time.Time `json:"assigned_at"`
}

type InventoryMaterialItem struct {
	ProjectMaterialID string  `json:"project_material_id"`
	MaterialName      string  `json:"material_name"`
	MaterialCode      *string `json:"material_code"`
	ProjectID         string  `json:"project_id"`
	ProjectName       string  `json:"project_name"`
	TotalQuantity     float64 `json:"total_quantity"`
	Unit              string  `json:"unit"`
}

type InventoryResponse struct {
	Employee  InventoryEmployeeSummary `json:"employee"`
	Assets    []InventoryAssetItem     `json:"assets"`
	Materials []InventoryMaterialItem  `json:"materials"`
}

// ── Transfer targets ───────────────────────────────────────────────────────────

type TransferTarget struct {
	ID        string `json:"id"`
	FirstName string `json:"first_name"`
	LastName  string `json:"last_name"`
	Role      string `json:"role"`
}

// ── Employee active projects ───────────────────────────────────────────────────

type InventoryProject struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

// ── Create transfer ────────────────────────────────────────────────────────────

// CreateTransferRequest is bound from the JSON body of POST /api/inventory/transfers.
// Exactly one of EmployeeAssetID or ProjectMaterialID must be set.
type CreateTransferRequest struct {
	ToEmployeeID         string   `json:"to_employee_id"`
	Type                 string   `json:"type"` // "asset" | "material"
	EmployeeAssetID      *string  `json:"employee_asset_id"`
	ProjectMaterialID    *string  `json:"project_material_id"`
	DestinationProjectID *string  `json:"destination_project_id"`
	Quantity             *float64 `json:"quantity"`
	Notes                string   `json:"notes"`
}

// ── Transfer list ──────────────────────────────────────────────────────────────

type TransferListItem struct {
	ID                string    `json:"id"`
	FromEmployeeID    string    `json:"from_employee_id"`
	FromEmployeeName  string    `json:"from_employee_name"`
	ToEmployeeID      string    `json:"to_employee_id"`
	ToEmployeeName    string    `json:"to_employee_name"`
	AssetType         string    `json:"asset_type"`
	ItemName          string    `json:"item_name"`
	Quantity          float64   `json:"quantity"`
	TransferredAt     time.Time `json:"transferred_at"`
	Notes             *string   `json:"notes"`
	TransferredByName string    `json:"transferred_by_name"`
}

type TransferPagination struct {
	Page       int `json:"page"`
	PerPage    int `json:"per_page"`
	Total      int `json:"total"`
	TotalPages int `json:"total_pages"`
}

type TransferListResponse struct {
	Data       []TransferListItem `json:"data"`
	Pagination TransferPagination `json:"pagination"`
}
